firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -840,6 +840,10 @@ extension BrowserViewController: WKNavigationDelegate {
     }
 
     func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        if let url = webView.url, !isToolbarRefactorEnabled {
+            urlBar.locationView.hasSecureContent = webView.hasOnlySecureContent
```

Comment:
@razvanlitianu I just noticed this is almost identical to the fix I implemented in https://github.com/mozilla-mobile/firefox-ios/pull/22160/files, however I believe we need to ensure we're checking whether the webView belongs to the current tab? I believe it's possible for a webview that is not part of the selected tab to still send a didFinish callback (which is why we have code further in this func that handles the snapshots), but in that case the SSL icon should not be updated.